pkg/validate/result_aggregator.go

```diff
@@ -0,0 +1,52 @@
+package validate
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// ValidationResultAggregator allows you to aggregate validation result and generate single error.
+// It's thread-safe.
+type ValidationResultAggregator struct {
+	issuesMsgs []string
+	issuesCnt  int
+	m          sync.RWMutex
+}
+
+// Report aggregates validation result. Can be used as wrapper:
+//   err = rs.Report(validator.ValidateParameters(ctx, ifaceSchemas, params))
+func (r *ValidationResultAggregator) Report(result ValidationResult, err error) error {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	if err != nil {
+		return err
+	}
+	r.issuesCnt += result.Len()
+	if err := result.ErrorOrNil(); err != nil {
```

Comment:
Isn't it better to just get all error messages from the `ValidationResult`? All you do inside `result.ErrorOrNil()` with these error messages you concatenate them:`errors.New(strings.Join(msgs, "\n"))`. 

Pros of such approach:
- Easier to understand the flow (right now it's too complicated in my opinion)
- No `issuesCnt` needed, you just use `len(issuesMsgs)`
- More control on formatting "sub-errors" from a single `ValidationResult` inside `ValidationResultAggregator`.

Reply:
discussed f2f currently it's too complex to be flattened.